example/lib/snapshot_scene.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:math' as math;
+import 'package:arkit_plugin/arkit_plugin.dart';
+import 'package:flutter/material.dart';
+import 'package:vector_math/vector_math_64.dart' as vector;
+
+class SnapshotScenePage extends StatefulWidget {
+  @override
+  _SnapshotScenePageState createState() => _SnapshotScenePageState();
+}
+
+class _SnapshotScenePageState extends State<SnapshotScenePage> {
+  ARKitController arkitController;
+
+  @override
+  void dispose() {
+    arkitController?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+      appBar: AppBar(
+        title: const Text('ARKit in Flutter'),
+      ),
+      floatingActionButton: FloatingActionButton(
+        child: Icon(Icons.camera_alt),
+        onPressed: () async {
+          try {
+            await _snapshot().then((imageProvider) {
```

Comment:
I think it might be more comfortable for less experienced developers to see this format instead of `then`:
```dart
final image = await _snapshot();
await Navigator.push(..., imageProvider: image);
```